src/imap.rs

```diff
@@ -991,23 +997,36 @@ impl Imap {
             std::thread::spawn(move || {
                 let &(ref lock, ref cvar) = &*v;
                 if let Some(ref mut session) = &mut *session.lock().unwrap() {
-                    let mut idle = match session.idle() {
-                        Ok(idle) => idle,
-                        Err(err) => {
-                            eprintln!("failed to setup idle: {:?}", err);
-                            return;
+                    loop {
+                        let res = match session.idle() {
+                            Ok(mut idle) => {
+                                // most servers do not allow more than ~28 minutes; stay clearly below that.
+                                // a good value that is also used by other MUAs is 23 minutes.
+                                // if needed, the ui can call dc_imap_interrupt_idle() to trigger a reconnect.
+                                // idle.set_keepalive(Duration::from_secs(23 * 60));
```

Comment:
rogue comment

Reply:
no, it's still a fitting comment, justifying the 23*60 constant.